src/groups/bmq/bmqp/bmqp_ackeventbuilder.cpp

```diff
@@ -117,10 +127,31 @@ const bdlbb::Blob& AckEventBuilder::blob() const
 
     // Fix packet's length in header now that we know it.  Following is valid
     // (see comment in reset).
-    EventHeader& eh = *reinterpret_cast<EventHeader*>(d_blob.buffer(0).data());
-    eh.setLength(d_blob.length());
+    EventHeader& eh = *reinterpret_cast<EventHeader*>(
+        d_blob_sp->buffer(0).data());
+    eh.setLength(d_blob_sp->length());
+
+    return *d_blob_sp;
+}
+
+bsl::shared_ptr<bdlbb::Blob> AckEventBuilder::blob_sp() const
```

Comment:
There are basically 2 common things that could be done in a similar situation.
- Rename `shared_ptr<...> blob_sp()` -> `shared_ptr<...> blob()` and remove `Blob& blob()`. It is okay since `bmqp` builders are not exposed to SDK clients. But there will be a difference with `bmqa` builders that return `Blob& blob()`. We don't enforce interface compatibility between `bmqp` and `bmqa` builders, just use `bmqp` ones as pointer with implementation.
- Show the meaningful name of the operation that return shared pointer, like `shared_ptr<> bookClientInfo()`. This is not applicable to our use-case.
So I will go on the 1st way.